metarecord/exporter/jhs/builder.py

```diff
@@ -0,0 +1,255 @@
+            action, bindings.TOIMENPIDELUOKKA_TEKSTI, "ActionType"
+        ),
+        _create_element_or_none_from_obj_attr(
+            action, bindings.TOIMENPIDELUOKKA_TARKENNE_TEKSTI, "TypeSpecifier"
+        ),
+        *records,
+        id=str(action.uuid),
+    )
+
+
+def _build_phase(phase: Phase):
+    actions = [_build_action(action) for action in phase.actions.all()]
+    return bindings.TOIMENPIDETIEDOT(
+        _create_element_or_none_from_obj_attr(
+            phase, bindings.TOIMENPIDELUOKKA_TEKSTI, "PhaseType"
+        ),
+        _create_element_or_none_from_obj_attr(
+            phase, bindings.TOIMENPIDELUOKKA_TARKENNE_TEKSTI, "TypeSpecifier"
+        ),
+        *actions,
+        id=str(phase.uuid),
+    )
+
+
+def _build_function(function: Function):
+    phases = [_build_phase(phase) for phase in function.phases.all()]
+    return bindings.LUOKKA(
+        bindings.LUOKITUSTUNNUS(function.get_classification_code()),
+        bindings.NIMEKE(
+            bindings.NIMEKE_KIELELLA(
+                bindings.NIMEKE_TEKSTI(function.get_name()), kieliKoodi="fi"
+            )
+        ),
+        bindings.KASITTELYPROSESSI_TIEDOT(
+            _build_restriction_info(function),
+            _build_retention_info(function),
+            _create_element_or_none_from_obj_attr(
+                function, bindings.TIETOJARJESTELMA_NIMI, "InformationSystem"
+            ),
+            *phases,
+            id=str(uuid.uuid4()),
+        ),
+        id=str(function.uuid),
+    )
+
+
+def _build_classification(classification: Classification):
+    try:
+        func = (
+            Function.objects.prefetch_related(
+                "phases", "phases__actions", "phases__actions__records"
+            )
+            .filter(classification__uuid=classification.uuid)
+            .latest_approved()
+            .get()
+        )
+    except Function.DoesNotExist:
+        # No functions found, return only the classification.
+        return bindings.LUOKKA(
+            bindings.LUOKITUSTUNNUS(classification.code),
+            bindings.NIMEKE(
+                bindings.NIMEKE_KIELELLA(
+                    bindings.NIMEKE_TEKSTI(classification.title),
+                    kieliKoodi="fi",
+                )
+            ),
+            id=str(classification.uuid),
+        )
+
+    return _build_function(func)
+
+
+def _build_tos_info():
+    return bindings.TOS_TIEDOT(
+        bindings.NIMEKE(
+            bindings.NIMEKE_KIELELLA(
+                bindings.NIMEKE_TEKSTI("Helsingin kaupungin Tiedonohjaussuunnitelma"),
+                kieliKoodi="fi",
+            )
+        ),
+        bindings.YHTEYSHENKILO_NIMI("Tiedonhallinta"),
+        bindings.TOS_VERSIO(TOS_VERSION),
+        bindings.TILA_KOODI("3"),
+        bindings.ORGANISAATIO_NIMI("Helsingin kaupunki"),
+        bindings.LISATIEDOT(
+            "JHS 191 XML {:%Y-%m-%d %H:%M%Z} {}".format(
+                datetime.now(tz=pytz.timezone(settings.TIME_ZONE)),
```

Comment:
Technically not in scope of the PR but caught my eye

This approach appears to be unique, which is mildly surprising. [https://www.google.com/search?q="datetime.now(tz=pytz.timezone(settings.TIME_ZONE))"](https://www.google.com/search?client=firefox-b-d&q=%22datetime.now%28tz%3Dpytz.timezone%28settings.TIME_ZONE%29%29%22)

I think it works OK.. at least with Europe/Helsinki, but standard approach would be to use django's `timezone.localtime(timezone.now())`

Reply:
This is the way. Will fix it.